recipe/conda_build_config.yaml

```diff
@@ -0,0 +1,4 @@
+MACOSX_DEPLOYMENT_TARGET:  # [osx and x86_64]
```

Comment:
What's the reason for increasing the deployment target? Can you please add a comment on why passing `_LIBCPP_DISABLE_AVAILABILITY` wasn't sufficient?

Reply:
XGBoost uses C++17 features such as `std::filesystem`. Is passing `_LIBCPP_DISABLE_AVAILABILITY` sufficient to enable C++17 features?

Reply:
@jakirkham Any inputs on this?

Reply:
Yes, that should be sufficient.

Reply:
Let me try removing `MACOSX_DEPLOYMENT_TARGET` flags and see if the build still succeeds.